sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_fixed_point.asciidoc

```diff
@@ -0,0 +1,501 @@
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|`ArbitraryPrecisionFixedPointINTEL`    | 5922
+|`OpFixedSqrtINTEL`                     | 5923
+|`OpFixedRecipINTEL`                    | 5924
+|`OpFixedRsqrtINTEL`                    | 5925
+|`OpFixedSinINTEL`                      | 5926
+|`OpFixedCosINTEL`                      | 5927
+|`OpFixedSinCosINTEL`                   | 5928
+|`OpFixedSinPiINTEL`                    | 5929
+|`OpFixedCosPiINTEL`                    | 5930
+|`OpFixedSinCosPiINTEL`                 | 5931
+|`OpFixedLogINTEL`                      | 5932
+|`OpFixedExpINTEL`                      | 5933
+|====
+
+
+== Modifications to the SPIR-V Specification Version 1.5
+
+After Section 3.16, add a new section "3.16b Quantization Modes" as follows:
+
+=== Quantization Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode        ^| Behavior
+| 0      | TRN          | Truncate towards -Inf
+| 1      | TRN_ZERO     | Truncate towards 0
+| 2      | RND          | Round towards +Inf
+| 3      | RND_ZERO     | Round towards 0
+| 4      | RND_INF      | Round towards +/-Inf
+| 5      | RND_MIN_INF  | Round towards -Inf
+| 6      | RND_CONV     | Round towards even
+| 7      | RND_CONV_ODD | Round towards odd
+|====
+
+After Section 3.16, add a new section "3.16c Overflow Modes" as follows:
+
+=== Overflow Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode     ^| Behavior
+| 0      | WRAP      | Drop the bits to the left of the MSB
+| 1      | SAT       | Saturate to the closest of MIN or MAX
+| 2      | SAT_ZERO  | Set to 0 on overflow
+| 3      | SAT_SYM   | For unsigned, treat as SAT. +
+
+For signed: a positive overflow will saturate at the maximum positive value, whereas a negative overflow will saturate to the negation of the maximum positive value, as opposed to the most negative value.
+|====
+
+=== Capability
+Modify Section 3.31, *Capability*, adding a row to the Capability table:
+
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5922 | *ArbitraryPrecisionFixedPointINTEL* +
+
+Enables arbitrary precision fixed-point math instructions
+| *INTELArbitraryPrecisionIntegers*
+|====
+
+=== Instructions
+
+In Section 3.32.13, *Arithmetic Instructions*, add the following instructions:
+
+[cols="11", width="100%"]
+|=====
+10+<|*OpFixedSqrtINTEL* +
+
+An `OpTypeInt` representing an arbitrary precision fixed point number (ac_fixed) is passed in as the _Input_ and the square root of the value is returned in _Result_.
+The behavior of this function is undefined for input values < 0.
+
+_Result Type_ is an `OpTypeInt` of width _rW_ and is the type of _Result_.
+
+_Input Type_ is an `OpTypeInt` of width _W_ and is the type of _Input_.
+
+_S_ is an `OpTypeBool` that indicates if the input and output are signed or unsigned.
+A value of `1` indicates that the two types are signed.
+Note that the operation does not use the signedness capabilities of `OpTypeInt`.
```

Comment:
Have you considered using this extension together with `Shader` capability?

I'm asking that, because according to [validation rules for `Kernel` capabilities](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_validation_rules_for_kernel_a_href_capability_capabilities_a), `Signedness` of `OpTypeInt` must always be 0: in `Kernel` SPIR-V, integer types do not hold information about their sign and each operation performed on integer types instructs how to represent them.

If you care only about `Kernel` capability, then you can implicitly declare it by `INTELArbitraryPrecisionIntegers` and add a general note that new fixed-point types doesn't hold information about the sign and that information is encoded into operations performed on new types like it is done in the main spec. I think that by adding such general note you can remove this line from description of all new instructions